src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -1500,32 +1500,19 @@ std::vector<ExpressionPtr> Visitor::visit(Parser::ExpressionListContext* ctx) {
 }
 
 // ____________________________________________________________________________________
-template <typename Context>
-Triples Visitor::parseTriplesConstruction(Context* ctx) {
+Triples Visitor::visit(Parser::ConstructTriplesContext* ctx) {
   auto result = visit(ctx->triplesSameSubject());
-  Context* subContext = [](Context* ctx) -> Context* {
-    if constexpr (std::is_same_v<Context, Parser::ConstructTriplesContext>) {
-      return ctx->constructTriples();
-    } else if constexpr (std::is_same_v<Context,
-                                        Parser::TriplesTemplateContext>) {
-      return ctx->triplesTemplate();
-    }
-  }(ctx);
-  if (subContext) {
-    auto newTriples = visit(subContext);
-    ad_utility::appendVector(result, std::move(newTriples));
+  if (ctx->constructTriples()) {
+    ad_utility::appendVector(result, visit(ctx->constructTriples()));
   }
   return result;
```

Comment:
This function was now only used in one place, so I inlined it. I am unfamiliar with the usage patterns of construct. Do you know how large such a construct can reasonably get in practice? @joka921

Reply:
Typically Construct queries are rather small, so we can leave this rule for now.